providers/base/units/dock/jobs.pxu

```diff
@@ -2355,85 +2355,535 @@ _description: Check dock MAC address pass-through function, please make sure fol
+_steps:
+    1. Connect a pair of headphones to the dock.
+    2. Go to the Sound settings and make sure the correct Output is selected.
+    3. Click the Test button to play a sound to your audio device.
+_verification:
+    Did you hear a sound through the headphones and did the sound play without any distortion, clicks or other strange noises from your headphones?
+_siblings:
+    [{"id": "dock/all-reboot-audio-playback-headphones",
+      "depends": "dock/all-reboot-audio-speaker-headphone-plug-detection"},
+     {"id": "dock/all-hotplug-audio-playback-headphones",
+      "depends": "dock/all-hotplug-audio-speaker-headphone-plug-detection"},
+     {"id": "dock/all-suspend-audio-playback-headphones",
+      "depends": "dock/all-suspend-audio-speaker-headphone-plug-detection"},
+     {"id": "dock/all-poweroff-audio-playback-headphones",
+      "depends": "dock/all-poweroff-audio-speaker-headphone-plug-detection"}]
+
+id: dock/all-init-audio-microphone-plug-detection
+category_id: dock-audio
+plugin: user-interact
+estimated_duration: 60.0
+_summary: Microphone recognized when plugged to the dock test
+requires:
+ device.category == 'AUDIO'
+ package.name == 'pulseaudio-utils'
+command: pulse_active_port_change.py sources
+_purpose:
+    Check that system detects a microphone being plugged in.
+    (Skip this test if the dock does not have a microphone connector)
+_steps:
+    1. Prepare a microphone with a standard 3.5mm jack.
+    2. Locate the microphone jack on the dock under test.
+       Keep in mind that it may be shared with the headphone jack.
+    3. Run the test (you have 30 seconds from now on)
+    4. Plug the microphone into the appropriate jack.
+    5. Unplug the device for subsequent tests.
+_verification:
+    Verification is automatic, no action is required.
+    The test times out after 30 seconds (and fails in that case).
+_siblings:
+    [{"id": "dock/all-reboot-audio-microphone-plug-detection"},
+     {"id": "dock/all-hotplug-audio-microphone-plug-detection"},
+     {"id": "dock/all-suspend-audio-microphone-plug-detection"},
+     {"id": "dock/all-poweroff-audio-microphone-plug-detection"}]
+
+plugin: user-interact-verify
+category_id: dock-audio
+id: dock/all-init-audio-alsa-record-playback-external
+estimated_duration: 30.0
+flags: also-after-suspend
+_summary: External microphone plugged to the dock to record sound test
+depends: dock/all-init-audio-microphone-plug-detection
+requires:
+    device.category == 'AUDIO'
+    package.name == 'alsa-base'
+    package.name == 'pulseaudio-utils'
+    package.name == 'gstreamer1.0-plugins-good' or package.name == 'gstreamer0.10-plugins-good'
+command:
+    audio_settings.py store --file="$PLAINBOX_SESSION_SHARE"/pulseaudio_settings
+    audio_settings.py set --device=pci --volume=50
+    alsa_record_playback.sh
+    EXIT_CODE=$?
+    audio_settings.py restore --file="$PLAINBOX_SESSION_SHARE"/pulseaudio_settings
+    exit $EXIT_CODE
```

Comment:
Same as above. You might even be able to refactor some of the code between the two tests.

Reply:
@pieqq  These commands between two jobs have slightly deference like line2801 and line 2867. And also these command are composed of several python and shell script from `bin/`. Would it be too much if I also make these comnmands into a file `bin/` floder?

Reply:
By the way, these are also the original way to write these two jobs.